contracts/core/mpl-rewards/v1/test/MplRewards.t.sol

```diff
@@ -1,157 +1,483 @@
+        assertTrue(owner.try_mplRewards_setPaused(address(rewardsContract), false));
+        assertTrue(!rewardsContract.paused());
 
-        // Set to unpaused
-        assertTrue(!fakeGov.try_setPaused(false));
-        assertTrue(     gov.try_setPaused(false));
+        stakingToken.mint(address(account1), 2);
 
-        assertTrue(!mplRewards.paused());
-        assertTrue(fay.try_withdraw(100 * WAD));
+        account1.erc2258_increaseCustodyAllowance(stakingToken, address(rewardsContract), 2);
+        assertTrue(account1.try_mplRewards_stake(address(rewardsContract), 2));
+        assertTrue(account1.try_mplRewards_withdraw(address(rewardsContract), 1));
+
+        owner.mplRewards_setPaused(rewardsContract, true);
+
+        account1.erc2258_increaseCustodyAllowance(stakingToken, address(rewardsContract), 1);
+        assertTrue(!account1.try_mplRewards_stake(address(rewardsContract), 1));
+        assertTrue(!account1.try_mplRewards_withdraw(address(rewardsContract), 1));
+    }
 
-        // Fez can stake
-        fez.increaseCustodyAllowance(address(mplRewards), 100 * WAD);
-        assertTrue(fez.try_stake(100 * WAD));
+    function test_rewardsSingleEpoch() external {
+        SomeAccount owner = new SomeAccount();
+        SomeERC2258 rewardToken = new SomeERC2258("RWT", "RWT");
+        SomeERC2258 stakingToken = new SomeERC2258("SKT", "SKT");
+        MplRewards rewardsContract = new MplRewards(address(rewardToken), address(stakingToken), address(owner));
+
+        uint256 totalRewardsInWad = 25_000 * WAD;
+        uint256 rewardsDuration = 30 days;
+
+        SomeAccount[] memory farmers = new SomeAccount[](2);
+        farmers[0] = new SomeAccount();
+        farmers[1] = new SomeAccount();
+
+        for (uint256 i; i < farmers.length; ++i) {
+            stakingToken.mint(address(farmers[i]), 100 * WAD);
+            farmers[i].erc2258_increaseCustodyAllowance(stakingToken, address(rewardsContract), 100 * WAD);
+        }
+
+        farmers[0].mplRewards_stake(rewardsContract, 10 * WAD);
+
+        rewardToken.mint(address(rewardsContract), totalRewardsInWad);
+
+        owner.mplRewards_setRewardsDuration(rewardsContract, rewardsDuration);
+        owner.mplRewards_notifyRewardAmount(rewardsContract, totalRewardsInWad);
+        
+        uint256 start = block.timestamp;
+
+        assertEq(rewardsContract.rewardRate(), uint256(totalRewardsInWad) / rewardsDuration);
+        assertEq(rewardToken.balanceOf(address(rewardsContract)), totalRewardsInWad);
+
+        /*** Farmer-0 time = 0 post-stake ***/
+        assertEq(rewardsContract.totalSupply(), 10 * WAD);
+        assertEq(rewardsContract.rewardPerTokenStored(), 0);
+        assertEq(rewardsContract.userRewardPerTokenPaid(address(farmers[0])), 0);
+        assertEq(rewardsContract.earned(address(farmers[0])), 0);
+        assertEq(rewardsContract.rewards(address(farmers[0])), 0);
+        assertEq(rewardToken.balanceOf(address(farmers[0])), 0);
```

Comment:
For these repetetive assertions I think it makes sense to use the 

```
    assertRewardsAccounting({
            account:                address(farmers[1]),                 // Account for accounting
            totalSupply:            2 * 10 * WAD,                 // Fay + Fez stake
            rewardPerTokenStored:   dTime1_rpt,                   // Not updated yet
            userRewardPerTokenPaid: dTime1_rpt,                   // Used so Fez can't do claims on past rewards
            earned:                 dTime2_rpt * 10,  // Fez has not claimed any rewards that have accrued during dTime2
            rewards:                0,                            // Not updated yet
            rewardTokenBal:         0                             // Not updated yet
        });
```
function

Reply:
Ya, I wasn't done refactoring this. I am not a fan of how it looks now yet either.